src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -265,17 +257,16 @@ void ParticipantConfiguration::xmlTagCallback(
     impl::PtrParticipant p(new impl::ParticipantState(name, _meshConfig));
     _participants.push_back(p);
   } else if (tag.getName() == TAG_PROVIDE_MESH) {
-    PRECICE_ASSERT(_dimensions != 0); // setDimensions() has been called
     std::string name = tag.getStringAttributeValue(ATTR_NAME);
 
     mesh::PtrMesh mesh = _meshConfig->getMesh(name);
     PRECICE_CHECK(mesh,
                   R"(Participant "{}" attempts to provide an unknown mesh "{}". <mesh name="{}"> needs to be defined first.)",
                   _participants.back()->getName(), name, name);
-
+    _dimensions = mesh->getDimensions();
+    PRECICE_ASSERT(_dimensions != 0);
```

Comment:
This seems wrong (more similar occasions below).
The `ParticipantConfiguration` class should no longer have a `_dimensions` as a participant have have several meshes with different dimensions.

Reply:
I removed the `_dimensions` from the `ParticipantConfiguration`. For the watchpoint, I directly got the dimensions of the respective mesh, as already implemented before (but without setting some member variable for the participant).